proto_task_queue/worker.py

```diff
@@ -159,9 +159,19 @@ def _process_message(self, message: pubsub_message.Message) -> None:
     args = registration.task_args_class()
     task.args.Unpack(args)
 
+    # Convert the task to a loggable string.
+    try:
+      task_string = self._task_to_string(task)
+    except Exception:  # pylint: disable=broad-except
+      logging.exception(
+          'Unable to convert task of type %s to a string for logging.',
+          full_name)
+      message.nack()
```

Comment:
Under what conditions could a task that previously raised on `_task_to_string` eventually succeed here? I'm wondering if this should be `message.ack()` instead.

Reply:
As far as this code is concerned, _task_to_string could be basically anything, including something that uses the network. Also, this code tries to preserve Cloud Pub/Sub's at-least-once delivery.

For the thing we plan to use this for internally, there's a really easy mistake to make that would cause it to fail reliably. Fixing that mistake and deploying would cause it to succeed reliably. There's also an alternate implementation of the same thing that I think wouldn't be susceptible to that easy mistake, but would use the network so it would be susceptible to transient failures. (I can say more about this on-corp.)

Reply:
Good enough for me. LGTM